sdk/maps/azure-maps-search/azure/maps/search/_search_client.py

```diff
@@ -0,0 +1,905 @@
+from ._generated.models import (
+    PointOfInterestCategory,
+    ReverseSearchCrossStreetAddressResult,
+    Polygon,
+)
+from .models import (
+    BoundingBox,
+    StructuredAddress,
+    SearchAddressResult,
+    SearchAlongRouteOptions,
+    SearchAddressBatchResult,
+    ReverseSearchAddressResult,
+    ReverseSearchAddressBatchProcessResult,
+)
+
+from ._shared import (
+    parse_geometry_input
+)
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+    from azure.core.polling import LROPoller
+
+
+# By default, use the latest supported API version
+class MapsSearchClient(MapsSearchClientBase):
+    """Azure Maps Search REST APIs.
+
+    :param credential: Credential needed for the client to connect to Azure.
+    :type credential: ~azure.core.credentials.TokenCredential or ~azure.core.credentials.AzureKeyCredential
+    :keyword str base_url: Supported Maps Services or Language resource base_url
+     (protocol and hostname, for example: 'https://<resource-name>.mapsservices.azure.com').
+    :keyword str client_id: Specifies which account is intended for usage with the Azure AD security model.
+     It represents a unique ID for the Azure Maps account.
+    :keyword api_version: The API version of the service to use for requests.
+     It defaults to the latest service version.
+     Setting to an older version may result in reduced feature compatibility.
+    :paramtype api_version: str
+    """
+
+    def __init__(
+        self,
+        credential, # type: Union[AzureKeyCredential, TokenCredential]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+
+        super().__init__(
+            credential=credential, **kwargs
+        )
+
+    @distributed_trace
+    def get_geometries(
+        self,
+        geometry_ids,  # type: List[str]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> List[Polygon]
+        """**Get Geometries**
+        The Get Geometries service allows you to request the geometry data such as a city or country
+        outline for a set of entities, previously retrieved from an Online Search request in GeoJSON
+        format. The geometry ID is returned in the sourceGeometry object under "geometry" and "id" in
+        either a Search Address or Search Fuzzy call.
+
+        :param geometry_ids: list of geometry UUIDs string, previously retrieved from an Online Search request.
+        :type geometry_ids: list[str]
+        :return: The result list of Polygon/Geometries.
+        :rtype: List[~azure.maps.search.models.Polygon]
+        """
+        polygon_result = self._search_client.list_polygons(
+            geometry_ids,
+            **kwargs
+        )
+        result = [] if not polygon_result else polygon_result.polygons
+        return result
+
+
+    @distributed_trace
+    def fuzzy_search(
+        self,
+        query,  # type: str
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> "SearchAddressResult"
```

Comment:
Why are we using Python 2-style syntax when Azure SDK no longer supports Python 2? We should be using Python 3's syntax for keyword only arguments and type hinting.